datastream-kafka-connector/src/test/java/com/linkedin/datastream/connectors/kafka/TestKafkaConnectorTask.java

```diff
@@ -170,7 +171,58 @@ public void testConsumeWithStartingOffset() throws Exception {
     DatastreamTaskImpl task = new DatastreamTaskImpl(Collections.singletonList(datastream));
     task.setEventProducer(datastreamProducer);
 
-    KafkaConnectorTask connectorTask = createKafkaConnectorTask(task);
+    KafkaConnectorTask connectorTask = createKafkaConnectorTaskWithAutoOffsetResetConfig(task, "earliest");
+
+    // validate auto.offset.reset config is overridden to none (given the start offsets)
+    Assert.assertEquals(connectorTask.getConsumerAutoOffsetResetConfig(), "none");
+
+
+    LOG.info("Sending third set of events");
+
+    //send 100 more msgs
+    produceEvents(_kafkaCluster, _zkUtils, topic, 1000, 100);
+
+    if (!PollUtils.poll(() -> datastreamProducer.getEvents().size() == 200, 100, POLL_TIMEOUT_MS)) {
+      Assert.fail("did not transfer 200 msgs within timeout. transferred " + datastreamProducer.getEvents().size());
+    }
+
+    connectorTask.stop();
+    Assert.assertTrue(connectorTask.awaitStop(CONNECTOR_AWAIT_STOP_TIMEOUT_MS, TimeUnit.MILLISECONDS),
+        "did not shut down on time");
+  }
+
+  @Test
+  public void testConsumeWithStartingOffsetAndResetStrategy() throws Exception {
+    String topic = "pizza1";
+    createTopic(_zkUtils, topic);
+
+    LOG.info("Sending first set of events");
+
+    //produce 100 msgs to topic before start
+    produceEvents(_kafkaCluster, _zkUtils, topic, 0, 100);
+    Map<Integer, Long> startOffsets = Collections.singletonMap(0, 100L);
```

Comment:
was the choice of 100L intentional for start timestamp?

Reply:
yes intentional, and it is a start offset, not timestamp.
we produce 100 events, set startOffsets to 100, produce 200 more and make sure we consume 200 only to verify start offset metadata took effect